dmscripts/helpers/supplier_data_helpers.py

```diff
@@ -93,7 +93,13 @@ def get_users_personalisations(self):
                 self.logger.info(
                     'Building user personalisations for supplier {}'.format(supplier_framework['supplierId'])
                 )
-            for user in supplier_framework['users']:
+            primary_email_address = supplier_framework.get('declaration', {}).get('primaryContactEmail')
+            if primary_email_address:  # if the primary email is also a user it will be overwritten in the output dict
+                if 'users' in supplier_framework:
```

Comment:
I think there will always be a non-empty `users` in supplier_framework (otherwise we've got a problem for that supplier!)

Would something like this be simpler instead?

```
for user in supplier_framework['users']:
    output.update(self.get_user_personalisation(user, supplier_framework))

primary_email_address = supplier_framework.get('declaration', {}).get('primaryContactEmail')
if primary_email_address:
    output.update(self.get_user_personalisation(primary_email_address, supplier_framework))
```
As you say, if the primary email address is already in `'users'` then the personalisation is the same, and it doesn't matter if we overwrite or not.

Reply:
OK, I can move the primary email after the rest of the users, but I'd rather skip overwriting, so the condition would be:
`if primary_email_address and primary_email_address not in output:`
What do you think?

Reply:
Yep, sounds good 👍